src/rgw/rgw_data_sync.cc

```diff
@@ -2441,6 +2442,33 @@ class RGWObjFetchCR : public RGWCoroutine {
           }
           return set_cr_error(retcode);
         }
+        
+        // notify that object has synced to this zone
```

Comment:
note that a call to `FetchRemoteObj` doesn't actually mean that we synced the object. it sends a GET request with the `If-Modified-Since` header so that it only transfers the object if it's newer than our local copy. if not, we just get `ERR_NOT_MODIFIED` and don't have to do anything

you can find some extra logic in `RGWAsyncFetchRemoteObj::_send_request()` for perf counters that tries to detect whether or not it actually transferred anything

Reply:
the main purpose of this feature is to let the client know that the object is synced to the zone.
if the zone has a more updated copy of the object, then there is probably no harm in sending the notification.

Reply:
thanks!
moving the notification code to `RGWAsyncFetchRemoteObj::_send_request()` would actually simplify the code.
it already has: `RGWObjectCtx`, `rgw::sal::Attrs attr`, `rgw::sal::RadosBucket` and `rgw::sal::RadosObject` available.

Reply:
I moved the notification code to `RGWAsyncFetchRemoteObj::_send_request()` and reverted changes to the coroutine, including the one for `need_more_info`
Commit: 6211ee4afcbce75521a6c377bd0bb26bccac519b